### ISSUE TITLE
Failed ./tests/tests from README in examples/README.md

### ISSUE THREAD

**Issue by fieldofnodes** [QuEST-Kit/QuEST#344]
## TLDR;
I have some failed tests when I run `./tests/tests` from inside the `build` directory. This is the suggested command to run from the `examples/README.md` file.

So I have downloaded QuEST.

I have run this test in the `build` folder, which I have made inside the `QuEST` root folder. (something like `$HOME/QuEST/build/`)

Running

```
cmake .. -DTESTING=ON
make
make test
```

Results in

```
128/128 Test #128: unitary ..............................   Passed    0.06 sec

100% tests passed, 0 tests failed out of 128

Total Test time (real) = 2260.87 sec
```

All good so far.

Then I run

`./tests/tests` from inside the `build` directory and I get this

```

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tests is a Catch v2.10.0 host application.
Run with -? for options

-------------------------------------------------------------------------------
mixTwoQubitKrausMap
  input validation
  trace preserving
-------------------------------------------------------------------------------
/home/fieldofnodes/Projects/QuEST/tests/test_decoherence.cpp:1080
...............................................................................

/home/fieldofnodes/Projects/QuEST/tests/utilities.cpp:572: FAILED:

===============================================================================
test cases:    128 |    127 passed | 1 failed
assertions: 689509 | 689508 passed | 1 failed
```

I checked the `QuEST/tests/test_decoherence.cpp:1080` file and this is the chunck.

```
SECTION( "trace preserving" ) {
    
    // valid Kraus map
    int numOps = GENERATE( range(1,16) );
    std::vector<QMatrix> matrs = getRandomKrausMap(2, numOps);
    ComplexMatrix4 ops[numOps];
    for (int i=0; i<numOps; i++)
        ops[i] = toComplexMatrix4(matrs[i]);
        
    // make only one of the ops at a time invalid
    ops[GENERATE_REF( range(0,numOps) )].real[0][0] = 0;
    REQUIRE_THROWS_WITH( mixTwoQubitKrausMap(qureg, 0,1, ops, numOps), Contains("trace preserving") );
}
```

I checked `QuEST/tests/utilities.cpp:572` which has this line: `DEMAND( areEqual(conjprod, iden) );` that is inside this process

```
QMatrix getRandomUnitary(int numQb) {
    DEMAND( numQb >= 1 );

    QMatrix matr = getRandomQMatrix(1 << numQb);

    for (size_t i=0; i<matr.size(); i++) {
        QVector row = matr[i];
        
        // compute new orthogonal row by subtracting proj row onto prevs
        for (int k=i-1; k>=0; k--) {

            // compute row . prev = sum_n row_n conj(prev_n)
            qcomp prod = 0;
            for (size_t n=0; n<row.size(); n++)
                prod += row[n] * conj(matr[k][n]);
                            
            // subtract (proj row onto prev) = (prod * prev) from final row
            for (size_t n=0; n<row.size(); n++)
                matr[i][n] -= prod * matr[k][n];
        }
    
        // compute row magnitude 
        qreal mag = 0;
        for (size_t j=0; j<row.size(); j++)
            mag += pow(abs(matr[i][j]), 2);
        mag = sqrt(mag);
        
        // normalise row
        for (size_t j=0; j<row.size(); j++)
            matr[i][j] /= mag;
    }
    
    // ensure matrix is indeed unitary 
    QMatrix conjprod = matr * getConjugateTranspose(matr);
    QMatrix iden = getIdentityMatrix(1 << numQb);
    
    // generating big unitary matrices is hard; if we fail, default to identity
    if ( numQb >= 3 && !areEqual(conjprod, iden) ) {
        
        matr = getIdentityMatrix(1 << numQb);
        conjprod = matr;
    }
    DEMAND( areEqual(conjprod, iden) );
    
    // return the new orthonormal matrix
    return matr;
}
```

**Comment by rrmeister** [comment]
Hi Jonathan,
I cannot reproduce this behaviour locally. Do you consistently get the same error when you run tests multiple times?
